integration/test/promtool_metrics_test.go

```diff
@@ -50,3 +58,85 @@ func checkMetrics(t *testing.T, port int) {
 		}
 	}
 }
+
+func TestFormatQueries(t *testing.T) {
+	grafana.VisitDashboards(
+		[]string{
+			"../../grafana/dashboards/cmode",
+			"../../grafana/dashboards/cmode-details",
+			"../../grafana/dashboards/storagegrid",
+		},
+		func(path string, data []byte) {
+			changeExpr(t, path, data)
+		},
+	)
+
+}
+
+func changeExpr(t *testing.T, path string, data []byte) {
+	var (
+		updatedData  []byte
+		notFormatted bool
+		errorStr     []string
+		err          error
+	)
+
+	copy(updatedData, data)
```

Comment:
This copy does not do what you think it does. It copies from source to destination, but it only copies as much into destination as will fit. In this case, that means zero will be copied since updateData has length == 0 when copy is called.

You could use `updatedData := slices.Clone(data)`